src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -28,11 +30,23 @@ const NavHeader = (props) => {
     isHeaderHide,
     topicDetail,
     setIsFollow,
-    isFollow
+    isFollow,
+    hasSearch,
+    searchText,
+    setSearchText,
+    setDiscoveryLoadingData,
+    isFocus,
+    setIsFocus,
+    fetchDiscoveryData,
+    onCancelToken,
+    placeholderText,
+    setIsFirstTimeOpen,
+    getSearchLayout
   } = props;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of new props to the `NavHeader` component increases its complexity. Consider breaking down the functionality into smaller, more manageable components if possible. However, without seeing the full context of how these props are used, it's hard to make a definitive suggestion.



<!-- This is an auto-generated comment by OSS CodeRabbit -->